### PR TITLE
chore(ui): migrate Modal to TypeScript

### PR DESCRIPTION
**Pull request by gjaskiewicz**
# Summary

migrate Modal to TypeScript

# Changes Made

- migrated Modal to TypeScript
- migrated ModalFooter to TypeScript

# Related Issues

- Issue 1: [link to issue](https://github.com/cloudoperators/juno/issues/520)

# Testing Instructions

1. `npm i`
2. `turbo test`

# Checklist

- [ ] I have performed a self-review of my code.
- [ ] I have commented my code, particularly in hard-to-understand areas.
- [ ] I have added tests that prove my fix is effective or that my feature works.
- [ ] New and existing unit tests pass locally with my changes.
- [ ] I have made corresponding changes to the documentation (if applicable).
- [ ] My changes generate no new warnings or errors.

### CODE REVIEW

File: packages/ui-components/src/components/Modal/Modal.test.tsx [hunk 1]
```diff
@@ -314,14 +316,14 @@ describe("Modal", () => {
     )
     const user = userEvent.setup()
     expect(screen.getByRole("dialog")).toBeInTheDocument()
-    waitFor(() => {
-      user.keyboard("{Tab}")
+    const _p = waitFor(async () => {
```
Comment (barsukov) [inline packages/ui-components/src/components/Modal/Modal.test.tsx:319]: I delete the variable and tests were green.

File: packages/ui-components/src/components/Modal/Modal.test.tsx [hunk 2]
```diff
@@ -297,8 +297,10 @@ describe("Modal", () => {
     )
     expect(screen.getByRole("dialog")).toBeInTheDocument()
     expect(screen.getByRole("textbox")).toBeInTheDocument()
-    waitFor(() => {
-      expect(screen.getByRole("textbox")).toHaveFocus()
+
+    const _p = waitFor(() => {
```
Comment (barsukov) [inline packages/ui-components/src/components/Modal/Modal.test.tsx:301]: We can remove this variable it works without it

**Review by barsukov — CHANGES_REQUESTED** [review]
Few little changes 🥇

File: packages/ui-components/src/components/Modal/Modal.stories.tsx [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+} & ModalProps
+
+const Template = ({ closeOnConfirm, ...args }: TemplateProps) => {
+  const [isOpen, setOpen] = useState(false)
+
+  const open = () => {
+    setOpen(true)
+  }
+  // eslint-disable-next-line no-unused-vars
+  const close = (event: any) => {
```
[the remark this answers is not in the document]
Comment (gjaskiewicz) [reply packages/ui-components/src/components/Modal/Modal.stories.tsx]: No - not needed

File: packages/ui-components/src/components/Modal/Modal.test.tsx (hunk 2 above)
Comment (gjaskiewicz) [reply packages/ui-components/src/components/Modal/Modal.test.tsx]: It complains about floating promise - I added lint ignore rule.

File: packages/ui-components/src/components/Modal/Modal.test.tsx (hunk 1 above)
Comment (gjaskiewicz) [reply packages/ui-components/src/components/Modal/Modal.test.tsx]: Deleted too, but added lint ignore rule

**Review by barsukov — APPROVED** [review]

**Pull request merged**